outlines/fsm/json_schema.py

```diff
@@ -32,12 +32,14 @@
 DATE = r'"(?:\d{4})-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])"'
 TIME = r'"(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z)?"'
 UUID = r'"[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"'
+URI = r'"(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)"'
```

Comment:
1) The `www.` is unnecessary

```
>>> pattern = r'"(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)"'
>>> re.match(pattern, '"http://www.example.com"')
<re.Match object; span=(0, 24), match='"http://www.example.com"'>
>>> pattern = r'"(https?:\/\/)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)"'
>>> re.match(pattern, '"http://www.example.com"')
<re.Match object; span=(0, 24), match='"http://www.example.com"'>
```

2) `@` can only appear once. You might separate the pattern so the domain portion is preceded by something like

`([-a-zA-Z0-9:%._\+~#=]+@)?` 

3) Could you please leave a commend or add documentation stating that URI only supports a subset of https://datatracker.ietf.org/doc/html/rfc3986 

4) `\b` isn't supported by `interegular`, you can simply remove this.